src/twisted/spread/jelly.py

```diff
@@ -489,25 +486,21 @@ def jelly(self, obj):
         objType = type(obj)
         if self.taster.isTypeAllowed(qual(objType).encode('utf-8')):
             # "Immutable" Types
-            if ((objType is bytes) or
-                (objType is int) or
-                (objType is long) or
-                (objType is float)):
+            if objType in (bytes, int, long, float):
                 return obj
-            elif objType is types.MethodType:
-                aSelf = obj.__self__ if _PY3 else obj.im_self
-                aFunc = obj.__func__ if _PY3 else obj.im_func
-                aClass = aSelf.__class__ if _PY3 else obj.im_class
+            elif isinstance(obj, types.MethodType):
+                aSelf = obj.__self__
+                aFunc = obj.__func__
+                aClass = aSelf.__class__
                 return [b"method", aFunc.__name__, self.jelly(aSelf),
                         self.jelly(aClass)]
             elif objType is unicode:
                 return [b'unicode', obj.encode('UTF-8')]
-            elif objType is type(None):
+            elif isinstance(obj, type(None)):
```

Comment:
This looks like a complicated way of expressing
```suggestion
            elif obj is None:
```